src/PrettyPrinter.ml

```diff
@@ -42,22 +42,25 @@ module Doc : DOC = struct
   let in_kwd = !^"in"
 
   let rec of_type = function
-    | Type.Unit -> unit_type
-    | Type.Base idT -> !^idT
-    | Type.Prod (t1, t2) -> parens (of_type t1 ^^ cross ^^ of_type t2)
-    | Type.Arr (dom, cod) -> parens (of_type dom ^^^ arrow ^^^ of_type cod)
-    | Type.Box t -> box_type ^^ of_type t
+    | Location.{ data = Type.Unit; _ } -> unit_type
+    | Location.{ data = Type.Base idT; _ } -> !^idT
+    | Location.{ data = Type.Prod (t1, t2); _ } ->
+        parens (of_type t1 ^^ cross ^^ of_type t2)
+    | Location.{ data = Type.Arr (dom, cod); _ } ->
+        parens (of_type dom ^^^ arrow ^^^ of_type cod)
+    | Location.{ data = Type.Box t; _ } -> box_type ^^ of_type t
```

Comment:
Match with `Location.{ data = Type.Unit; _ }` at the beginning of the function. Try to keep the patch minimal.